src/main/java/uk/gov/companieshouse/authcodenotification/service/AuthCodeNotificationService.java

```diff
@@ -48,4 +47,16 @@ private String getOverseasEntityEmail(String requestId, String companyNumber, Ma
         return email;
     }
 
+    private String getCompanyName(String requestId, String companyNumber, Map<String, Object> logMap) throws ServiceException {
+        String companyName = privateDataRetrievalService.getCompanyProfile(requestId, companyNumber).getCompanyName();
+
+        if (StringUtils.isBlank(companyName)) {
+            var e = new ServiceException("Null or empty company name found");
+            ApiLogger.errorContext(requestId, "Failed to retrieve a valid company number", e, logMap);
+            throw e;
```

Comment:
I think you should create/instantiate the ServiceException on this line and throw it here too. Good practice. Will try and dig out a link explaining why (if required). When logging the error in the line above, don't pass any exception in (or null).

Reply:
Separate line.